column_buffer.go

```diff
@@ -1263,7 +1272,11 @@ func (col *floatColumnBuffer) WriteFloats(values []float32) (int, error) {
 
 func (col *floatColumnBuffer) WriteValues(values []Value) (int, error) {
 	var model Value
-	col.writeValues(makeArrayValue(values, unsafe.Offsetof(model.u64)), columnLevels{})
+	if cpu.IsBigEndian {
+		col.writeValues(makeArrayValue(values, unsafe.Offsetof(model.u64)+4), columnLevels{})
+	} else {
+		col.writeValues(makeArrayValue(values, unsafe.Offsetof(model.u64)), columnLevels{})
+	}
```

Comment:
Rather than repeating the if/else branches everywhere, could we factor the offset value into a constant with a different value depending on the target architecture?

```go
//go:build s390
...
const offsetOfModelU64 = unsafe.Offsetof(model.u64)
```
```go
//go:build !s390
...
const offsetOfModelU64 = unsafe.Offsetof(model.u64)
```
```go
col.writeValues(makeArrayValue(values, offsetOfModelU64), columnLevels{})
```

Reply:
The constant won't be redeclared if you set the build tags properly so that only one of the files gets included depending on whether we're building for s390x or not (maybe the build tag was wrong in my example, please double-check).

We don't need the model to be declared as a local variable, for example:
```go
const offsetOfU64 = unsafe.Offsetof(Value{}.u64)
```
As long as the expression in `Offsetof` can be evaluated at compile time, the result is a constant expression.

This would greatly simplify the structure of the code, it seems worth making the change.